src/main/java/sirius/kernel/commons/Files.java

```diff
@@ -147,8 +154,39 @@ public static String getFilenameWithoutExtension(@Nullable String path) {
     }
 
     /**
-     * If the given file is not null and exists, tries to delete that file and logs when a file cannot be deleted. T
-     * his is useful for error reporting and to diagnose why a file cannot be deleted.
+     * Determines if the given path is hidden.
+     * <p>
+     * A path is considered hidden if it starts with a dot.
+     *
+     * @param fileOrDirectoryName the path to check
+     * @return <tt>true</tt> if the path is hidden, <tt>false</tt> otherwise
+     */
+    public static boolean isConsideredHidden(@Nullable String fileOrDirectoryName) {
+        if (Strings.isEmpty(fileOrDirectoryName)) {
+            return false;
+        }
+        return fileOrDirectoryName.startsWith(".");
+    }
+
+    /**
+     * Determines if the given path is a metadata file.
+     * <p>
+     * A metadata file is a file which is not part of the actual content but rather contains metadata or is used by
+     * the operating system or other tools.
+     *
+     * @param path the path to check
+     * @return <tt>true</tt> if the path is a metadata file that is listed in the METADATA_FILES list, <tt>false</tt> otherwise
+     */
+    public static boolean isConsideredMetadata(@Nullable String path) {
+        if (Strings.isEmpty(path)) {
+            return false;
+        }
+        return streamPath(path).anyMatch(METADATA_FILES::contains);
```

Comment:
This should now go back to .startsWith, since it only expects a target file/dir right?
PS: test cases must be than adjusted.